API/src/Modules/XpSystems/Momentum.XpSystems.Infrastructure/Repositories/XpSystemRepository.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Marten;
+using System.Threading.Tasks;
+using Momentum.XpSystems.Core.Models;
+using Momentum.XpSystems.Core.Repositories;
+
+namespace Momentum.XpSystems.Infrastructure.Repositories
+{
+    public class XpSystemRepository : IXpSystemRepository
+    {
+        private readonly IDocumentStore _store;
+
+        public XpSystemRepository(IDocumentStore store)
+        {
+            _store = store;
+        }
+
+        public async Task<XpSystem> CreateOrUpdate(XpSystem model)
+        {
+            using var session = _store.LightweightSession();
+
+            session.Store(model);
+
+            await session.SaveChangesAsync();
+
+            return model;
```

Comment:
One last thing, XpSystem should inherit `TimeTrackedModel`, and set the respective CreatedAt/UpdatedAt properties.

Look at [GenericTimeTrackedRepository](https://github.com/momentum-mod/website/blob/rewrite/API/src/Framework/Momentum.Framework.Infrastructure/Repositories/GenericTimeTrackedRepository.cs#L21-L46) to see how that is done before. 

Since add/update are done in the same method, this logic may work:

```cs
if (model.CreatedAt == default) 
{
    model.CreatedAt = DateTime.UtcNow;
} 
else
{
    model.UpdatedAt = DateTime.UtcNow;
}
```